tests/workflow/interfaces/run/test_autograd_run.py

```diff
@@ -0,0 +1,143 @@
+class TestAutogradRun:
+    """Test the 'autograd' interface run function integrates well for both forward and backward execution"""
+
+    def test_run(self, device, config, shots, seed):
+        """Test execution of tapes on 'autograd' interface."""
+        device = get_device(device, seed=seed)
+        config = replace(config, interface="autograd")
+
+        def cost(a, b):
+            ops1 = [qml.RY(a, wires=0), qml.RX(b, wires=0)]
+            tape1 = qml.tape.QuantumScript(ops1, [qml.expval(qml.PauliZ(0))], shots=shots)
+
+            ops2 = [qml.RY(a, wires="a"), qml.RX(b, wires="a")]
+            tape2 = qml.tape.QuantumScript(ops2, [qml.expval(qml.PauliZ("a"))], shots=shots)
+
+            resolved_config = _resolve_execution_config(
+                config, device, [tape1, tape2], TransformProgram()
+            )
+            inner_tp = _setup_transform_program(TransformProgram(), device, resolved_config)[1]
+            return run([tape1, tape2], device, resolved_config, inner_tp)
+
+        a = pnp.array(0.1, requires_grad=True)
+        b = pnp.array(0.2, requires_grad=False)
+
+        with device.tracker:
+            res = cost(a, b)
+
+        assert len(res) == 2
+
+        if getattr(config, "grad_on_execution", False):
+            assert device.tracker.totals["execute_and_derivate_batches"] == 1
+        else:
+            assert device.tracker.totals["batches"] == 1
+        assert device.tracker.totals["executions"] == 2
+
+        if not shots.has_partitioned_shots:
+            assert res[0].shape == ()
+            assert res[1].shape == ()
+
+        assert qml.math.allclose(res[0], np.cos(a) * np.cos(b), atol=atol_for_shots(shots))
+        assert qml.math.allclose(res[1], np.cos(a) * np.cos(b), atol=atol_for_shots(shots))
+
+    def test_scalar_jacobian(self, device, config, shots, seed):
+        """Test scalar jacobian calculation"""
+        device = get_device(device, seed=seed)
+        config = replace(config, interface="autograd")
+
+        def cost(a):
+            tape = qml.tape.QuantumScript([qml.RY(a, 0)], [qml.expval(qml.PauliZ(0))], shots=shots)
+            resolved_config = _resolve_execution_config(config, device, [tape], TransformProgram())
+            inner_tp = _setup_transform_program(TransformProgram(), device, resolved_config)[1]
+            return run([tape], device, resolved_config, inner_tp)[0]
+
+        a = pnp.array(0.1, requires_grad=True)
+        if shots.has_partitioned_shots:
+            res = qml.jacobian(lambda x: qml.math.hstack(cost(x)))(a)
+        else:
+            res = qml.jacobian(cost)(a)
+            assert res.shape == ()  # pylint: disable=no-member
+
+        # compare to standard tape jacobian
+        tape = qml.tape.QuantumScript([qml.RY(a, wires=0)], [qml.expval(qml.PauliZ(0))])
+        tape.trainable_params = [0]
+        tapes, fn = qml.gradients.param_shift(tape)
+        expected = fn(device.execute(tapes))
```

Comment:
```suggestion
        expected = qml.math.cos(a)
```
Convenient to use `RX`/ `RY` + expval Z.  Allows you to jut compare to a cosin.

Reply:
Oh I guess we were already comparing the results to `-np.sin(a)`.  Do we still need the param shift based comparison then?

Reply:
I'm not sure. This is from the `execute` tests. 😄 I guess not since the point of the test is to make sure the gradient using `run` is aligned with the analytical result.

Reply:
[sc-72838]